### ISSUE TITLE
Create an Event website template

### ISSUE THREAD

**Issue by chrisguindon** [EclipseFdn/events.eclipse.org#12]
We are planning to host small events under events.eclipse.org.

The goal is to create a generic template to allow us to create a new website in less than 2 hours of work!

@ChristieWittEclipse is planning to help us out with this initiative! We are thinking of creating 3-4 variations so that we can switch between for each new event.

Let's use this bug to discuss requirements!

**Comment by chrisguindon** [comment]
All event website will have the following section

- Jumbotron
  - Title
  - Slogan (optional)
  - Date
- About us
  - 2-3 paragraphs about the event
  - Registration link
- Agenda
- Speakers and/or Program Committee (Optional)
- Sponsors (Optional)

We plan on leveraging the featured footer section with a call to action to see all events on events.eclipse.org.

**Comment by chrisguindon** [comment]
The Virtual IoT and Edge Native Day event website is a good example of what we are trying to build here: https://iot.eclipse.org/eclipse-virtual-iot-2020/

**Comment by chrisguindon** [comment]
The technical requirements for this project are being track via EclipseFdn/hugo-solstice-theme:
https://github.com/EclipseFdn/hugo-solstice-theme/issues/119

@autumnfound is currently migrating event components from the jakartone.org template to our Hugo theme which will allow us to do most of what we want to do here!

**Comment by chrisguindon** [comment]
We plan on creating the SAM IoT 2020 #11 and the AI in EU Virtual Workshop #13 website with this new template.

**Comment by ChristieWittEclipse** [comment]
@chrisguindon +1, let me know when @autumnfound has the components staged in the Hugo theme, and then I can provide image assets as per the specs of that theme. I think we can start with 3 template variations.

**Comment by chrisguindon** [comment]
@ChristieWittEclipse +1

I plan on using the preview url'S  for both SAM IoT 2020 #11 (https://deploy-preview-11--eclipsefdn-events.netlify.app/sam-iot-2020/) and  the AI in EU Virtual Workshop #13 event website to show you the components in action.

If you need something now, the Virtual IoT and Edge Native Day is a great starting point of a simple event webpage. 

The only difference will be the colors on events.eclipse.org will be the eclipse.org look and feel:
https://iot.eclipse.org/eclipse-virtual-iot-2020/

**Comment by ChristieWittEclipse** [comment]
@chrisguindon  +1

**Comment by chrisguindon** [comment]
This is the preview url for the AI in EU Virtual Workshop event website:
https://deploy-preview-16--eclipsefdn-events.netlify.app/2020/ai-eu-virtual-workshop/

There is not much right now but we will update as we get new components in our hugo base theme.

**Comment by ChristieWittEclipse** [comment]
@chrisguindon thanks, I have the first generic template with Thabang and Shanda for review by noon today. I should be able to provide you assets by EOD.

**Comment by chrisguindon** [comment]
Great! Is it an extension of the work you did for Sam IoT or is it a new design?

**Comment by ChristieWittEclipse** [comment]
@chrisguindon It is a mix of the SAM IoT website and the Virtual IoT and Edge Native Day website templates/designs. I received a few revisions so will update you later today!

**Comment by ChristieWittEclipse** [comment]
@chrisguindon I've attached the approved mockups for the first generic virtual event template, with both light and dark variations as we discussed. The image assets are also attached. I will work on the second and third generic template images within the next week.

[Generic Template 1 for Web Team.zip](https://github.com/EclipseFdn/events.eclipse.org/files/4593237/Generic.Template.1.for.Web.Team.zip)

**Comment by chrisguindon** [comment]
@ChristieWittEclipse Thank you Christie! 

All good for the additional variations! I think we can use this desgign for #17 and #16

**Comment by ChristieWittEclipse** [comment]
@chrisguindon +1 that's great!

**Comment by chrisguindon** [comment]
I created an issue https://github.com/EclipseFdn/jquery-eclipsefdn-api/issues/16 to discuss our maintenance problem with the featured footer.

As described in https://github.com/EclipseFdn/events.eclipse.org/issues/12#issuecomment-623724477, we are thinking of leveraging the featured footer to display a call to action to view all our events as a temporary solution while we work on an easier way to manage this section for all of our sites.

If we want to use the featured footer for something else, we should consider adding a link to "view all events" somewhere in our template.  We often get this request each time we spin up a new Event website. 

For example, https://github.com/jakartaee/jakartaone.org/issues/74

**Comment by ChristieWittEclipse** [comment]
+1, I think for this initial template we will keep the footer as is, but I will leave this decision to @shanda-eclipse. As I have time closer to Q3, I can design the generic footer that will work across all of the different working group websites and the eclipse.org website.

**Comment by chrisguindon** [comment]
> +1, I think for this initial template we will keep the footer as is, but I will leave this decision to @shanda-eclipse.

+1 I am fine with keeping our current footer which is  "Sign up to our newsletter" 

![Screen Shot 2020-05-07 at 10 51 07 AM](https://user-images.githubusercontent.com/893942/81309202-acda6a00-9050-11ea-81c1-baedf2318281.png)

**Comment by shanda-eclipse** [comment]
> > +1, I think for this initial template we will keep the footer as is, but I will leave this decision to @shanda-eclipse.
> 
> +1 I am fine with keeping our current footer which is "Sign up to our newsletter"
> 
> ![Screen Shot 2020-05-07 at 10 51 07 AM](https://user-images.githubusercontent.com/893942/81309202-acda6a00-9050-11ea-81c1-baedf2318281.png)

+1

**Comment by ChristieWittEclipse** [comment]
@chrisguindon FYI – I will be holding on designing the other 2 templates until I have the capacity.

**Comment by chrisguindon** [comment]
@ChristieWittEclipse +1 Thanks for the update!

**Comment by chrisguindon** [comment]
I am done implementing the look and feel that @ChristieWittEclipse  created for our Virtual Events:
https://deploy-preview-16--eclipsefdn-events.netlify.app/2020/open-source-ai-workshop/

Please note that EVERY section on this page is optional. The best way to describe our architecture is to think about lego blocks. Each section can be configured to have a white, gray, or dark background + we can control the order of each section.

Finally, we can also re-use the same section more than once. 

For example,  in the link that I shared with you, we are using our user_display component for both the speaker and program committee section. We are using Gael as an example, but you can image 2 separate lists of people for each section…

**Comment by ChristieWittEclipse** [comment]
@chrisguindon we had a few tweaks to this template that I missed on the call:

**About the Event Section**
-Add “Co-organized by:” text and logo placeholder below the “About the event” description as per the initial mockup.
![Screen Shot 2020-05-14 at 4 53 19 PM](https://user-images.githubusercontent.com/58480647/82330992-85b95c00-99b1-11ea-8508-fd290fc6c18a.png)

**Speakers section** 
-Center the speaker name & title text
 
**Agenda section** 
-Change Title from “Sessions” to “Agenda” to correspond to the Jumbotron button
-In the Legend, change a grey circle to be dark grey #808285
 
**Program Committee section** 
-Center the PC name & title text
 
**Sponsors section**
-Make Heading the same size as the other section headings
-Add "Become a Sponsor" button below the sponsor logo section (linking to contact-us form)

**Comment by chrisguindon** [comment]
@ChristieWittEclipse,
> @chrisguindon we had a few tweaks to this template that I missed on the call:

As discussed last week, the event page is to prove/show that our template is configurable.

For example, if we want the content to be center-align, we would simply add the .text-center class to the wrapping div!

> **About the Event Section**
> -Add “Co-organized by:” text and logo placeholder below the “About the event” description as per the initial mockup.
> ![Screen Shot 2020-05-14 at 4 53 19 PM](https://user-images.githubusercontent.com/58480647/82330992-85b95c00-99b1-11ea-8508-fd290fc6c18a.png)

I won't add this section on the event page for Open Source AI Workshop since we can already see this in action via https://events.eclipse.org/2020/sam-iot/
 
> 
> **Speakers section**
> -Center the speaker name & title text

Done! I added .text-center class to the wrapping div.

> 
> **Agenda section**
> -Change Title from “Sessions” to “Agenda” to correspond to the Jumbotron button

I created a change in our base theme:
https://github.com/EclipseFdn/hugo-solstice-theme/commit/5ae7ccd4e378e838460c4bffa73e9ace2019c0d8

 You will only be able to see the change when we decide to release a new version of our base theme which will probably later this week!

> -In the Legend, change a grey circle to be dark grey #808285

Done! This was a typo in our agenda.yml file.

> 
> **Program Committee section**
> -Center the PC name & title text

Done! I added .text-center class to the wrapping div.

> **Sponsors section**
> -Make Heading the same size as the other section headings

We are using h2 for each section heading. They are all the same size (font-size: 34px;) :) 

> -Add "Become a Sponsor" button below the sponsor logo section (linking to contact-us form)

Thabang requested that we **Create a contact form for event sponsorships** via infrazilla:
https://foundation.eclipse.org/infrazilla/show_bug.cgi?id=3735

I would recommend that we discuss the details for this on the bug that Thabang created! I think this is a good idea but I have a few questions about how this is supposed to work. 

I will need some answers before we implement the solution!
